shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -831,6 +843,13 @@ object Types {
         case tycon: TypeProxy                       => tycon.superType.applyIfParameterized(args)
         case _                                      => AnyType
 
+    override def translucentSuperType(using Context): Type = tycon match
+      case tycon: TypeRef if tycon.symbol.isOpaqueTypeAlias =>
+        tycon.translucentSuperType.applyIfParameterized(args)
```

Comment:
not verified but this seems to throw some red flags for if the aliased type has different type parameters

Reply:
If it has different type parameters, the problem would be malformed. Wouldn't it?

Reply:
I was thinking of an some thing like ```opaque type Foo[A, B] = List[A]``` then wouldn't that apply two params to list, but since this is what dotty does I won't question

Reply:
No, when you have that definition, in TASTy it's actually a
```scala
opaque type Foo = [A, B] =>> List[A]
```
with a `TypeLambda`. So you apply the type lambda, which substitutes `A` and `B` for what you apply it to (discarding `B` in the process).